modules/integration/src/test/java/org/apache/fluo/integration/impl/OracleIT.java

```diff
@@ -318,14 +322,8 @@ public void threadFailoverTest() throws Exception {
     oserver3.close();
   }
 
-  private void sleepWhileConnected(OracleServer oserver) throws InterruptedException {
-    while (oserver.isConnected()) {
-      Thread.sleep(100);
-    }
-  }
-
-  private void sleepUntilConnected(OracleServer oserver) throws InterruptedException {
-    while (!oserver.isConnected()) {
+  private void sleepWhile(Supplier<Boolean> condition) throws InterruptedException {
+    while (condition.get()) {
```

Comment:
I like how you changed this to a supplier, but I also liked the readability of the `sleepUntil` method name.

You could add it back as:

```java
private void sleepUntil(Supplier<Boolean> condition) throws InterruptedException {
  sleepWhile(() -> !condition.get());
}
```

And then, all your statements would like like one of these two:

```java
sleepWhile(oserver::isConnected());
sleepUntil(oserver::isConnected());
```

That would remove all the excess `() ->` stuff a bit, too.

Reply:
I like that suggestion, much more readable.